packages/@react-spectrum/dnd/docs/dragAndDrop.mdx

```diff
@@ -0,0 +1,256 @@
+from the list after a drop operation. A drop target that only allows `copy` operations, such as a file upload drop zone, would be able to return `copy` from its `getDropOperation` and communicate that
+to your draggable collection's `onDragEnd` handler, letting it know that it shouldn't remove the dragged items from the list. Alternatively, a drop target that allows `move` operations, like in the
+[example](#example) below, would return `move` from its `getDropOperation` and thus inform your draggable collection to remove the dragged items from it list.
+
+You may be wondering why we didn't simply import the drag and drop hooks within the components themselves. The reason behind this is to reduce the bundle size of
+the component for users who don't need the drag and drop support for their component.
+
+The following list shows which components currently support drag and drop:
+-  [ListView](ListView.html#drag-and-drop)
+
+## Example
+
+The example below illustrates how one would use `useDragHooks` and `useDropHooks` to enable dragging files from one ListView to another. For the first ListView, we want to make the rows draggable and
+have the dragged rows removed from the list upon a successful drop. To accomplish this, we first want to specify the types of data associated with each item via the array returned by `getItems`. In this case,
+we add a `text/plain` type that matches the text value associated with each item. The item's type is also included with the item's `id` mapped to it. We also provide a `onDragEnd` event handler to `useDragHooks`
+that removes the dragged items from the list only upon a successful drop operation.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+import Folder from '@spectrum-icons/workflow/Folder';
+import {Item, ListView} from '@react-spectrum/list';
+import {Text} from '@adobe/react-spectrum';
+import {useDragHooks} from '@react-spectrum/dnd';
+import {useListData} from '@react-stately/data';
+
+function DraggableList(props) {
+  let {
+    items,
+    ...otherProps
+  } = props;
+  let list = useListData({
+    initialItems: items || [
+      {id: 'a', textValue: 'Adobe Photoshop', type: 'file'},
+      {id: 'b', textValue: 'Adobe XD', type: 'file'},
+      {id: 'c', textValue: 'Documents', type: 'folder'},
+      {id: 'd', textValue: 'Adobe InDesign', type: 'file'},
+      {id: 'e', textValue: 'Utilities', type: 'folder'}
+    ]
+  });
+
+  let getItems = (keys) => [...keys].map(key => {
+    let item = list.getItem(key);
+    return {
+      'text/plain': item.textValue,
+      [item.type]: item.id
+    };
+  });
+
+  let onDragEnd = (e) => {
+    if (e.dropOperation === 'move') {
+      list.remove(...e.keys);
+    }
+  };
+  let dragHooks = useDragHooks({
+    getItems,
+    onDragEnd
```

Comment:
I think it might be easier to read if the event handlers/functions are inline here.